doc/tdshim_spec.md

```diff
@@ -0,0 +1,619 @@
+  // The first 512 bytes are boot_param. (zero page)
+  // The entrypoint is start address of loaded 64bit Linux kernel
+  //   plus 0x200
+  PayloadImageTypeBzImage,
+
+  // Payload Binary is VMM loaded vmLinux, follow Linux boot protocol.
+  // The entrypoint is defined at HOB_PAYLOAD_INFO_TABLE.Entrypoint.
+  PayloadImageTypeRawVmLinux,
+
+
+} PAYLOAD_IMAGE_TYPE;
+
+#pragma pack ()
+```
+
+### Payload Loading
+
+The VMM shall load the payload such as Linux kernel (bzImage or vmlinux) and optional driver (initrd.img) with payload command line parameter into TD private memory if they are supported in TD metadata. The payload may or might not be included in the binary and the payload may or might not be measured in MRTD based upon different use cases. If the payload image is not measured in MRTD, then it shall be measured to RTMR[1].
+
+NOTE: The VMM does not need create the payload boot params, such as the Linux boot zero page, but just create a payload boot command line parameter (a string, please refer to https://www.kernel.org/doc/html/v5.0/admin-guide/kernel-parameters.html, and https://man7.org/linux/man-pages/man7/bootparam.7.html). It is TD Shim that setup the required OS kernel boot_params (the zero page). See Part III for detail.
+
+
+## Part III - TD Shim / Guest Payload Interface
+
+### Boot Protocol
+
+If the payload follows Linux Boot Protocol, TD Shim shall follow 64bit boot protocol defined in https://www.kernel.org/doc/Documentation/x86/boot.txt to setup the boot_params (the zero page), then jump to the 64-bit payload entry point according to the TD Payload Info GUID Extension HOB.
+
+The bootstrap processor state is below:
+ * CPU must be in 64-bit mode with paging enabled.
+ * The range with setup_header.init_size from start address of loaded Linux kernel, zero page and Linux kernel command line parameter buffer get identical mapping.
+ * A GDT must be loaded with below configuration
+    * CS=0x10, 4G flat segment, with execute/read permission.
+    * DS=0x18, 4G flat segment, with read/write permission.
+    * ES=DS.
+    * SS=DS.
+ * Interrupt must be disabled.
+ * RSI must hold the base address of the struct boot_params.
+
+If the payload does not follow Linux Boot Protocol, TD Shim shall follow X64 calling convention to call the kernel entrypoint and pass a new payload HOB as the first parameter.
+ * For PE/COFF image, MSVC X64 calling convention must be used. RCX must hold the payload HOB address.
+ * For ELF image, System-V ABI AMD64 calling convention must be used. RDI must hold the payload HOB address.
+ * All rest are same as above.
+
+### UEFI Interface
+
+TD Shim shall only NOT support UEFI interface. If the TD guest requires to boot an UEFI OS, the TD Shim may load a special UEFI payload to support UEFI interface and UEFI OS kernel boot.
+
+### ACPI Table
+
+TD Shim shall only support static ACPI tables. TD Shim shall not report any ACPI table that contains the ASL such as DSDT or SSDT.
```

Comment:
Just trying to understand the rationale behind this one as this is quite a strong requirement which will require the VMM to expose a different set of ACPI tables.
And about Kata, that means we won't be able to perform PCI hotplug through ACPI.

Reply:
Yes, that's true. There's no CPU/memory/PCI hotplug with current spec. The goal is to disable ASL in guest.

Reply:
@sboeuf It is indeed my understanding that we can't have hotplug / unplug with confidential platforms, also for security reasons. See https://docs.google.com/presentation/d/1U6EJmhNLs9wZPUPWqI1dCMacHoxmk86MDirPGkWs54Y/edit#slide=id.ge7a49b5a89_1_597 (from page 19 on).